src/map/Pins.js

```diff
@@ -31,26 +34,109 @@
 
         this.pinsPaths = this._pinsTransform.selectAll(".data").data(this.visible() ? this.data() : [], function (d) { return d[0]; });
         var context = this;
-        this.pinsPaths.enter().append("path")
-            .attr("class", "data")
-            .attr("d", function (d) {
-                return "M 0,0 C -2,-20 -10,-22 -10,-30 A 10,10 0 1,1 10,-30 C 10,-22 2,-20 0,0 z M -2,-30";
-            })
-            .call(this._selection.enter.bind(this._selection))
-            .on("click", function (d) {
-                context.click(context.rowToObj(d), "weight", context._selection.selected(this));
-            })
-        ;
-        this.pinsPaths
-            .style("fill", function (d) {
-                return d[2] && d[2].fillColor ? d[2].fillColor : context.pinColor();
-            })
-        ;
+        this.pinsPaths.enter().append("g").each(function(d) {
+            if (!d[2] || d[2].pinShape !== "none") {
+                d3.select(this).append("path")
+                    .attr("class", "data")
+                    .attr("d", function (d) {
+                        var pinShape = context.pinShape();
+                        if (d[2] && d[2].pinShape) {
+                            pinShape = d[2].pinShape;
+                        }
+                        switch (pinShape) {
+                            case "circle":
+                                return "m-24,-31c0,10.8335 6.8925,20.0592 16.53,23.53l8.47,8.47l8.47,-8.47c9.6375,-3.4708 16.53,-12.6965 16.53,-23.53c0,-13.8073 -11.1927,-25 -25,-25c-13.8073,0 -25,11.1927 -25,25z";
+                            case "square":
+                                return "m-23.53906,-54.02734l0,40c0,3.0458 0.9541,4 4,4l10,0l10,10l10,-10l10,0c3.0459,0 4,-0.9542 4,-4l0,-40c0,-3.0458 -0.9541,-4 -4,-4l-40,0c-3.0459,0 -4,0.9542 -4,4z";
```

Comment:
Can we round these up to integers - it will simplify the process of publishing various properties for each (like width, height etc.).
(If you look at the following to see how D3 defines "symbols", you will see what I am aiming for):  https://github.com/mbostock/d3/blob/a10b22654c9275da1050e1513bbd834646c0bde3/src/svg/symbol.js#L49-L100